api/api_full.go

```diff
@@ -738,6 +738,37 @@ type FullNode interface {
 
 	NodeStatus(ctx context.Context, inclChainStatus bool) (NodeStatus, error) //perm:read
 
+	// MethodGroup: Eth
+	// These methods are used for Ethereum-compatible JSON-RPC calls
+	//
+	// EthAccounts will always return [] since we don't expect Lotus to manage private keys
+	EthAccounts(context.Context) ([]EthAddress, error) //perm:read
+	// EthBlockNumber returns the height of the latest (heaviest) TipSet
+	EthBlockNumber(ctx context.Context) (EthInt, error) //perm:read
+	// EthGetBlockTransactionCountByNumber returns the number of messages in the TipSet
+	EthGetBlockTransactionCountByNumber(context.Context, string) (EthInt, error) //perm:read
+	// EthGetBlockTransactionCountByHash returns the number of messages in the TipSet
+	EthGetBlockTransactionCountByHash(context.Context, string) (EthInt, error) //perm:read
+
+	EthGetBlockByHash(ctx context.Context, blkHash string, fullTxInfo bool) (EthBlock, error)                       //perm:read
+	EthGetBlockByNumber(ctx context.Context, blkNumHex string, fullTxInfo bool) (EthBlock, error)                   //perm:read
+	EthGetTransactionByHash(ctx context.Context, txHash string) (EthTx, error)                                      //perm:read
+	EthGetTransactionCount(ctx context.Context, sender string, blkOpt string) (EthInt, error)                       //perm:read
```

Comment:
If you also implement [UnmarshalJSON](https://pkg.go.dev/encoding/json#Unmarshaler) on Eth* types you should be able to use them for parameters as well

Reply:
addressed comment